buildSrc/src/main/kotlin/com/saveourtool/processbuilder/DetektConfiguration.kt

```diff
@@ -19,32 +19,35 @@ fun Project.configureDetekt() {
     apply<DetektPlugin>()
     configure<DetektExtension> {
         config = rootProject.files("detekt.yml")
+        basePath = rootDir.canonicalPath
         buildUponDefaultConfig = true
     }
-    @Suppress("GENERIC_VARIABLE_WRONG_DECLARATION")
-    val reportMerge: TaskProvider<ReportMergeTask> = rootProject.tasks.named<ReportMergeTask>("mergeDetektReports") {
-        input.from(
-            tasks.withType<Detekt>().map { it.sarifReportFile }
-        )
-        shouldRunAfter(tasks.withType<Detekt>())
-    }
-    tasks.withType<Detekt>().configureEach {
-        reports.sarif.required.set(true)
-        finalizedBy(reportMerge)
-    }
 }
 
 /**
  * Register a unified detekt task
  */
 fun Project.createDetektTask() {
-    tasks.register("detektAll") {
+    val detektAllTask = tasks.register("detektAll") {
         allprojects {
             this@register.dependsOn(tasks.withType<Detekt>())
         }
     }
 
     tasks.register("mergeDetektReports", ReportMergeTask::class) {
+        mustRunAfter(detektAllTask)
         output.set(buildDir.resolve("detekt-sarif-reports/detekt-merged.sarif"))
     }
+
+    @Suppress("GENERIC_VARIABLE_WRONG_DECLARATION")
+    val reportMerge: TaskProvider<ReportMergeTask> = rootProject.tasks.named<ReportMergeTask>("mergeDetektReports") {
```

Comment:
it's the same task as created on prev line `tasks.register("mergeDetektReports", ReportMergeTask::class) {`
why not merged them?